service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,116 @@
+        List<Element<DocumentSocialWorkOther>> listOfOtherDocs = setUpdatedByAndDateAndTimeForDocuments(
+                caseData.getOtherSocialWorkDocuments(), caseDataBefore.getOtherSocialWorkDocuments());
+
+        Document socialWorkChronologyDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getSocialWorkChronologyDocument(), caseDataBefore.getSocialWorkChronologyDocument());
+        Document socialWorkStatementDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getSocialWorkStatementDocument(), caseDataBefore.getSocialWorkStatementDocument());
+        Document socialWorkAssessmentDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getSocialWorkAssessmentDocument(), caseDataBefore.getSocialWorkAssessmentDocument());
+        Document socialWorkCarePlanDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getSocialWorkCarePlanDocument(), caseDataBefore.getSocialWorkCarePlanDocument());
+        Document socialWorkEvidenceTemplateDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getSocialWorkEvidenceTemplateDocument(),
+                caseDataBefore.getSocialWorkEvidenceTemplateDocument());
+        Document thresholdDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getThresholdDocument(), caseDataBefore.getThresholdDocument());
+        Document checklistDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getChecklistDocument(), caseDataBefore.getChecklistDocument());
+        Document courtBundleDocument = setUpdatedByAndDateAndTimeForDocuments(
+            caseData.getCourtBundle(), caseDataBefore.getCourtBundle()
+        );
+
+        Map<String, Object> updatedCaseData = new HashMap<>();
+
+        updatedCaseData.put("documents_socialWorkOther", listOfOtherDocs);
+        updatedCaseData.put("documents_socialWorkChronology_document", socialWorkChronologyDocument);
+        updatedCaseData.put("documents_socialWorkStatement_document", socialWorkStatementDocument);
+        updatedCaseData.put("documents_socialWorkAssessment_document", socialWorkAssessmentDocument);
+        updatedCaseData.put("documents_socialWorkCarePlan_document", socialWorkCarePlanDocument);
+        updatedCaseData.put("documents_socialWorkEvidenceTemplate_document", socialWorkEvidenceTemplateDocument);
+        updatedCaseData.put("documents_threshold_document", thresholdDocument);
+        updatedCaseData.put("documents_checklist_document", checklistDocument);
+        updatedCaseData.put("courtBundle", courtBundleDocument);
+
+        return updatedCaseData;
+    }
+
+    public <T extends DocumentMetaData> List<Element<T>> setUpdatedByAndDateAndTimeForDocuments(
+        List<Element<T>> listOfCurrentDocs,
+        List<Element<T>> listOfOldDocs) {
+
+        Predicate<Element<T>> doesNotContainInOldDocs = doc -> listOfOldDocs != null && !listOfOldDocs.contains(doc);
```

Comment:
I'd move this into it's own private method and name it 'doesNotEqualDocument'

Reply:
I would like to understand why you say that please? 

and name i think to me what i have is more appropriate.

Reply:
`doesNotContainDocument` is just as, even more appropriate perhaps. Removal of the word 'old' is what i'm getting at here. It's just a method, it should not matter whether document is old or new.